cli/smartcontract/smart_contract.go

```diff
@@ -819,6 +841,90 @@ func contractDeploy(ctx *cli.Context) error {
 	return nil
 }
 
+// contractUpdate updates an existing smart contract.
+func contractUpdate(ctx *cli.Context) error {
+	var (
+		paramsStart   = 0
+		args          []string
+		nefBytes      []byte
+		manifestBytes []byte
+		nefParam      any
+		manifestParam any
+	)
+	args = ctx.Args().Slice()
+	if len(args) < 1 {
+		return cli.Exit(errNoScriptHash, 1)
+	}
+	contractHash, err := flags.ParseAddress(args[0])
+	if err != nil {
+		return cli.Exit(fmt.Errorf("invalid contract hash '%s': %w", args[0], err), 1)
+	}
+	paramsStart++
+
+	if ctx.String("in") != "" {
+		_, nefBytes, err = readNEFFile(ctx.String("in"))
+		if err != nil {
+			return cli.Exit(fmt.Errorf("failed to read .nef file: %w", err), 1)
+		}
+		nefParam = nefBytes
+	}
+	if ctx.String("manifest") != "" {
+		_, manifestBytes, err = readManifest(ctx.String("manifest"), contractHash)
+		if err != nil {
+			return cli.Exit(fmt.Errorf("failed to read manifest file: %w", err), 1)
+		}
+		manifestParam = manifestBytes
+	}
+	if nefBytes == nil && manifestBytes == nil {
+		return cli.Exit(fmt.Errorf("either manifest or .nef required"), 1)
+	}
+
+	params := []any{nefParam, manifestParam}
+
+	signOffset, data, err := cmdargs.ParseParams(args[paramsStart:], true)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("unable to parse 'data' parameter: %w", err), 1)
+	}
+	if len(data) > 1 {
+		return cli.Exit("'data' should be represented as a single parameter", 1)
+	}
+
+	acc, w, err := options.GetAccFromContext(ctx)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("can't get sender address: %w", err), 1)
+	}
+	defer w.Close()
+
+	cosigners, exitErr := cmdargs.GetSignersFromContext(ctx, paramsStart+signOffset)
+	if exitErr != nil {
+		return exitErr
+	}
+	if len(cosigners) == 0 {
+		cosigners = []transaction.Signer{{
+			Account: acc.Contract.ScriptHash(),
+			Scopes:  transaction.CalledByEntry,
+		}}
+	}
+	if len(data) != 0 {
+		params = append(params, data[0])
+		return invokeWithArgs(ctx, acc, w, contractHash, "update", params, cosigners)
+	}
+	params = append(params, nil)
```

Comment:
Deserves a comment. Something like "Try NEP-22 compatible format first, invoke without data if failed". You may rephrase, but it should be clear why do we call it twice.